### ISSUE TITLE
amp-carousel cropping the item DOM height in desktop UA

### ISSUE THREAD

**Issue by uskay** [ampproject/amphtml#18352]
## What's the issue?

When using amp-carousel in **desktop UA** && adding carousel item as non-`amp-img` DOM, since `i-amphtml-slides-container` has `overflow-x: auto !important`, `height: 100% !important` will be calculated minusing the scroll bar's height, which will end up to shorter container height and item being cropped.

**An image like this...**
![abc](https://cdn.glitch.com/98e724d1-70f9-4ce1-91d7-23acbcacb55f%2FScreen%20Shot%202018-09-25%20at%2021.45.30.png?1537900168946)
**...will end up showing like this in `amp-carousel` when the items are surrounded by \<a\> tag (\<a href="/"\>\<amp-img\>\</amp-img\>\</a\>)**
![abc](https://cdn.glitch.com/98e724d1-70f9-4ce1-91d7-23acbcacb55f%2FScreen%20Shot%202018-09-26%20at%2002.24.54.png?1537900153035)
**Live example: AliExpress**
![abc](https://cdn.glitch.com/98e724d1-70f9-4ce1-91d7-23acbcacb55f%2FScreen%20Shot%202018-09-26%20at%2002.24.37.png?1537900160177)

## How do we reproduce the issue?
Repro page: https://amp-carousel-size-test.glitch.me/
1. Add carousel items as non-`amp-img` DOM e.g. \<div\>, \<a\>...
2. Open the page in desktop UA

*If using Mac's trackpad, scrollbar is automatically hidden so you will need to use a mouse or reproduce by setting [always to Show scroll bars](http://osxdaily.com/2011/08/03/show-scroll-bars-mac-os-x-lion/).
*If you remove `padding-bottom: 20px` from `i-amphtml-slides-container`, it's very clear that the scroll bar height is breaking the container height.
![abc](https://cdn.glitch.com/98e724d1-70f9-4ce1-91d7-23acbcacb55f%2FpHqzsuoYW9f.png?1537900463746)

## Suggestion
What do you think about simply inheriting `i-amphtml-slides-container`'s height from parent rather than `height: 100%`?
https://github.com/uskay/amphtml/commit/a99fc21023b380036c87037cab196129ef2fb192#diff-99d5d6d12e0d5695a02b76fbd2ef5353
![abc](https://cdn.glitch.com/98e724d1-70f9-4ce1-91d7-23acbcacb55f%2FvhaPz4XOku6.png?1537900129471)

## What browsers are affected?
All browsers (which will show scroll bars on `overflow-x: auto`)

## Which AMP version is affected?
Powered by AMP ⚡ HTML – Version 1809142227020

**Comment by uskay** [comment]
TIL, it is also defined here that height percentage will be calculated based on the `padding box` height when the element itself is `absolute`
https://www.w3.org/TR/CSS22/visudet.html#the-height-property

I guess the definition `padding box` here is the same as `Element.clientHeight`?

> inner height of an element in pixels, including padding **but not the horizontal scrollbar height**, border, or margin
https://developer.mozilla.org/en-US/docs/Web/API/Element/clientHeight

Which makes a lot of sense regarding this issue. 
- having `overflow-x: auto` might lead to showing a horizontal scroll bar e.g. using a mouse in desktop UA
- having `i-amphtml-slides-container` as `position: absolute` && `height: 100%` will calculate the element height subtracting the **horizontal scrollbar height**  (vs the item's actual size should be the same as the parent \<amp-carousel\>)
- And having `overflow-y: hidden` will crop the item based on the calculated height above

Adding @aghassemi to this thread.

**Comment by choumx** [comment]
Thanks for the detailed report and investigation!

**Comment by aghassemi** [comment]
Let's verify in the new carousel.
